### PR TITLE
Use pattern matching.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Prremptive merging.  Let me know if there's anything you'd like me to change.

**Comment by Pilchie** [comment]
Tagging @dotnet/roslyn-ide for review.

File: src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs [hunk 1]
```diff
@@ -37,14 +37,14 @@ internal static class GoToDefinitionHelpers
             // We can't go to the definition of the alias, so use the target type.
 
             var solution = project.Solution;
-            if (symbol is IAliasSymbol)
+            if (alias != null)
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs]: Combine with the other `alias != null` check above?

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs [hunk 2]
```diff
@@ -603,11 +602,10 @@ public override SyntaxNode CreateFieldDeclaration(IFieldSymbol field, CodeGenera
             {
                 return null;
             }
+            // we will ignore the method if the associated property can be generated.
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs]: Odd there's a blank line after this. Was this automated?

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**